edgelet/edgelet-http-keyservice/src/error.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use failure::{Backtrace, Context, Fail};
+use std::fmt::{self, Display};
+
+use crate::IntoResponse;
+use hyper::{Response, Body, StatusCode};
+use keyservice::models::ErrorResponse;
+use hyper::header::{CONTENT_LENGTH, CONTENT_TYPE};
+use log::error;
+
+#[derive(Debug)]
+pub struct Error {
+    inner: Context<ErrorKind>,
+}
+
+#[derive(Clone, Debug, Fail)]
+pub enum ErrorKind {
```

Comment:
Give IS, KS, CS and MR are REST servers, can some of the code in error.rs and lib.rs be reused?

Reply:
It isn't clear what code could be reused. The error messages are different and the libs have different routes defined.